cli/command/system/dial_stdio.go

```diff
@@ -34,10 +34,17 @@ func runDialStdio(dockerCli command.Cli) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to open the raw stream connection")
 	}
-	connHalfCloser, ok := conn.(halfCloser)
-	if !ok {
+
```

Comment:
defer conn.Close here?

Reply:
Seems plausible and WFM in my simple cases, so I've pushed. I think this was just a preexisting leak, but it also help allay concerns about the `nopCloseReader` here too.

Would be good to get this tested with your ssh:// use case though.